src/VisualStudio/Core/Def/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs

```diff
@@ -32,45 +33,50 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
         private readonly VisualStudioSuppressionFixService _suppressionFixService;
         private readonly VisualStudioDiagnosticListSuppressionStateService _suppressionStateService;
         private readonly IUIThreadOperationExecutor _uiThreadOperationExecutor;
         private readonly IDiagnosticAnalyzerService _diagnosticService;
         private readonly ICodeActionEditHandlerService _editHandlerService;
-        private readonly IWpfTableControl? _tableControl;
         private readonly IAsynchronousOperationListener _listener;
 
+        private IWpfTableControl? _tableControl;
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioDiagnosticListTableCommandHandler(
+            IThreadingContext threadingContext,
             SVsServiceProvider serviceProvider,
             VisualStudioWorkspace workspace,
             IVisualStudioSuppressionFixService suppressionFixService,
-            IVisualStudioDiagnosticListSuppressionStateService suppressionStateService,
+            VisualStudioDiagnosticListSuppressionStateService suppressionStateService,
             IUIThreadOperationExecutor uiThreadOperationExecutor,
             IDiagnosticAnalyzerService diagnosticService,
             ICodeActionEditHandlerService editHandlerService,
             IAsynchronousOperationListenerProvider listenerProvider)
         {
+            _threadingContext = threadingContext;
             _workspace = workspace;
             _suppressionFixService = (VisualStudioSuppressionFixService)suppressionFixService;
-            _suppressionStateService = (VisualStudioDiagnosticListSuppressionStateService)suppressionStateService;
+            _suppressionStateService = suppressionStateService;
             _uiThreadOperationExecutor = uiThreadOperationExecutor;
             _diagnosticService = diagnosticService;
             _editHandlerService = editHandlerService;
-
-            var errorList = serviceProvider.GetService(typeof(SVsErrorList)) as IErrorList;
-            _tableControl = errorList?.TableControl;
             _listener = listenerProvider.GetListener(FeatureAttribute.ErrorList);
         }
 
-        public void Initialize(IServiceProvider serviceProvider)
+        public async Task InitializeAsync(IAsyncServiceProvider serviceProvider, CancellationToken cancellationToken)
         {
+            var errorList = await serviceProvider.GetServiceAsync<SVsErrorList, IErrorList>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
+            _tableControl = errorList?.TableControl;
+
             // Add command handlers for bulk suppression commands.
-            var menuCommandService = (IMenuCommandService)serviceProvider.GetService(typeof(IMenuCommandService));
+            var menuCommandService = (IMenuCommandService?)await serviceProvider.GetServiceAsync(typeof(IMenuCommandService)).ConfigureAwait(false);
```

Comment:
Use the newer helper here, so there'se no risk of that cast being on the wrong thread?